### DISCUSSION TITLE
Arc A770: ImportError: libmkl_sycl.so.3: cannot open shared object file:

### DISCUSSION THREAD

**Discussion by HubKing** [vladmandic/sdnext#2860] [Q&A] [answered]
I get the following error on Arch Linux. I searched the Internet and found an issue https://github.com/intel/intel-extension-for-pytorch/issues/417 . Is it the same issue? I don't understand this well, but is this a problem of Intel's OneAPI Base Kit itself, or a problem of the Arch packager, or SD.Next? 

For what it's worth, Blender has also been crashing with `error while loading shared libraries: libsycl.so.6: cannot open shared object file: No such file or directory` for some time. They both have "sycl". 

```
./webui.sh --use-ipex
Create and activate python venv
Launching ipexrun launch.py...
/home/user/diffusion/automatic/venv/lib/python3.11/site-packages/torchvision/io/image.py:13: UserWarning: Failed to load image Python extension: ''If you don't plan on using image functionality from `torchvision.io`, you can ignore this warning. Otherwise, there might be something wrong with your environment. Did you have `libjpeg` or `libpng` installed before building `torchvision` from source?
  warn(
Traceback (most recent call last):
  File "/home/user/diffusion/automatic/venv/bin/ipexrun", line 5, in <module>
    from intel_extension_for_pytorch.launcher import main
  File "/home/user/diffusion/automatic/venv/lib/python3.11/site-packages/intel_extension_for_pytorch/__init__.py", line 93, in <module>
    from .utils._proxy_module import *
  File "/home/user/diffusion/automatic/venv/lib/python3.11/site-packages/intel_extension_for_pytorch/utils/_proxy_module.py", line 2, in <module>
    import intel_extension_for_pytorch._C
ImportError: libmkl_sycl.so.3: cannot open shared object file: No such file or directory

```

**Comment by Disty0** [comment, marked as answer]
What version of ipex do you have? Run webui.sh with `DISABLE_IPEXRUN=1 ./webui.sh --reinstall --use-ipex` and it will install the latest one and necessary mkl libs.

**Reply by HubKing** [reply]
Running `--reinstall` (without `--use-ipex`, because it caused the same error as above) fixed the problem. I had run things like "--upgrade" or "--requirements" but those had not fixed the problem. I don't know what version of "IPEX" I had, but the system packages were all up-to-date, and I had ran "git pull" to update this project, so I don't know why the `webuse.sh` did not update that package.

**Reply by Disty0** [reply]
You probably had IPEX 2.0 while SDNext installs MKL and DPCPP for IPEX 2.1.
Also SDNext installs MKL and DPCPP on it's venv so you don't have to install OneAPI BaseKit.

Probably ipexrun was failing with `--use-ipex`. You can also export `DISABLE_IPEXRUN=1` to disable it.
Updated my original message to reflect this.

`--upgrade` only updates SDNext.
Updating PyTorch and IPEX needs `--reinstall` argument.
